src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -254,7 +264,13 @@ public Task<SourceText> ReadTextAsync(CancellationToken cancellationToken)
                     // of a page fault. Therefore, if we're going to be blocking a thread, we should
                     // just block one thread and do the whole thing at once vs. a fake "async"
                     // implementation which will continue to requeue work back to the thread pool.
-                    return Task.Factory.StartNew(() => ReadText(cancellationToken), cancellationToken, TaskCreationOptions.None, TaskScheduler.Default);
+                    if (_service._workspaceThreadingService is { IsOnMainThread: true })
+                    {
+                        await Task.Yield().ConfigureAwait(false);
```

Comment:
so, i have no problem with the new code.  But i would be interested to know what was wrong with the old code that was a problem enough to need fixing.  was it just that we were creating too many tasks?

Reply:
> This change should improve the ability to investigate issues like [AB#1526707](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1526707) by reducing the amount of operation decoupling that occurs on paths that call ReadTextAsync.

ah ok.